app/assets/javascripts/discourse/app/components/plugin-outlet.js

```diff
@@ -79,12 +80,22 @@ export default class PluginOutletComponent extends GlimmerComponentWithDeprecate
     return result;
   }
 
-  get connectors() {
-    return renderedConnectorsFor(
+  @bind
+  getConnectors({ hasBlock }) {
+    const connectors = renderedConnectorsFor(
       this.args.name,
       this.outletArgsWithDeprecations,
       this.context
     );
+    if (connectors.length > 1 && hasBlock) {
+      // eslint-disable-next-line no-console
+      console.error(
+        `Multiple connectors were registered for the ${this.args.name} outlet. Using the first.`,
+        connectors.map((c) => c.humanReadableName)
+      );
+      return [connectors[0]];
```

Comment:
This is interesting, but can be confusing too. For example, if a theme defines connectors for A and B and then another theme defines connectors for B and C. There is a chance that the first theme will have only connector A visible and the second theme will have B and C visible, which may seem like the first theme is only partially enabled?

Also, I believe if the two themes define the same A and B, then there is a chance you will see connector A from theme #1 and connector B from theme #2.

Reply:
Do you suggest any solution to this?